dotnet/csharp.bzl

```diff
@@ -407,6 +417,104 @@ def csharp_configure():
       actual = "@local_config_csharp//:csc_bin",
   )
 
+def _nuget_dep_impl(ctx):
+  inputs = set(ctx.files.srcs)
+  return struct(
+    name = ctx.label.name,
+    target_type = ctx.attr._target_type,
+    out = inputs,
+    dlls = inputs,
+    transitive_dlls = set([]),
+  )
+
+nuget_dep = rule(
+  implementation = _nuget_dep_impl,
+  attrs = _NUGET_ATTRS,
+)
+"""Bundles up a nuget package to be used as a dependency for dotnet builds.
+
+Args:
+  name: A unique name for this rule.
+  srcs: dotnet `.dll` assemblies.
+"""
+
+def _nuget_package_impl(repository_ctx):
+  # figure out the output_path
+  pkg_name = repository_ctx.attr.pkg_name
+  output_dir = repository_ctx.path("")
+
+  # assemble our nuget command
+  nuget_cmd = [
+    repository_ctx.attr.nuget_bin_path,
+    "install",
+    "-Version", repository_ctx.attr.pkg_version,
+    "-OutputDirectory", output_dir,
+  ]
+  # add the sources from our source list to the command
+  for source in repository_ctx.attr.nuget_package_source:
+    nuget_cmd += ["-Source", source]
+
+  # Lastly we add the nuget package name.
+  nuget_cmd += [repository_ctx.attr.pkg_name]
+  # execute nuget download.
+  repository_ctx.execute(nuget_cmd)
+  # TODO(jeremy): report errors if there were any
+
+  tpl_file = Label("@io_bazel_rules_dotnet//dotnet:NUGET_BUILD.tpl")
+  # add the BUILD file
+  repository_ctx.template(
+    "BUILD",
+    tpl_file,
+    {"%{pkg_name}": repository_ctx.name,
+     "%{output_dir}": "%s" % output_dir})
+
+nuget_package = repository_rule(
+  implementation=_nuget_package_impl,
+  #local=False,
+  attrs={
+    # Location of the nuget exe
+    "nuget_bin_path":attr.string(default="/usr/local/bin/nuget"),
+    # Sources to download the nuget packages from
+    "nuget_package_source":attr.string_list(),
+    # The name of the nuget package
+    "pkg_name":attr.string(mandatory=True),
+    # The version of the nuget package
+    "pkg_version":attr.string(mandatory=True),
+    # content of the BUILD file for this external resource.
+  })
+"""Fetches a nuget package as an external dependency.
+
+This rule is a repository rule and is only usable in WORKSPACE files.
+due to some current limitations of repository_rules it does require you to
+tell it where your nuget.exe is located. You may want to manage that binary
+in your repository as a result.
+
+Args:
+  nuget_package_source: list of sources to use for nuget package feeds.
+  pkg_name: name of the nuget package.
+  pkg_version: version of the nuget package (e.g. 0.1.2)
+"""
+
+NUNIT_BUILD_FILE = """
```

Comment:
Is this change intentional? I moved the BUILD file for nunit into a standalone BUILD file in f30f2f80984cea9cb53aa46ceacbc7c4046c2f89.